src/views/modal_handler.rs

```diff
@@ -72,28 +82,48 @@ impl DisplayModal {
             // Display warning for unsaved changes
             ModalMessage::UnsavedChangesExitModal => {
                 self.show_modal = true;
-                self.is_warning = true;
-                self.title = "Unsaved Changes".to_string();
-                self.body =
-                    "You have unsaved changes. Do you want to exit without saving?".to_string();
+                self.modal_type = Some(ModalType::Warning {
```

Comment:
No big deal - but I thought you would use modal_type != None, to replace "show_modal"...